packetbeat/protos/amqp/amqp_structs.go

```diff
@@ -0,0 +1,201 @@
+package amqp
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"time"
+)
+
+type AmqpMethod func(*AmqpMessage, []byte) (bool, bool)
+
+const (
+	TransactionsHashSize = 2 ^ 16
+	TransactionTimeout   = 10 * 1e9
+)
+
+//layout used when a timestamp must be parsed
+const (
+	AMQPTIMELAYOUT = "January _2 15:04:05 2006"
+)
+
+//Frame types and codes
+const (
+	METHOD    = 1
+	HEADER    = 2
+	BODY      = 3
+	HEARTBEAT = 8
+)
+
+const (
+	FRAME_END_OCTET byte = 206
+)
+
+//Methods
+const (
+	CONNECTION = 10
```

Comment:
consider using ypes for constants. Makes methodMap in amqp module more clear. Plus don't put 'module-private' in all upper case. More idiomatic:

```
type codeClass uint16

const (
    connectionCode codeClass = 10
    ...
)

type codeMethod uint16

const (
   connectionStartCode codeMethod = 10
)

```